base/docs/Docs.jl

```diff
@@ -675,4 +675,19 @@ function hasdoc(binding::Docs.Binding, sig::Type = Union{})
 end
 
 
+"""
+    undocumented_names(mod::Module; all=false)
+
+Return an array of undocumented symbols in `module` (that is, lacking docstrings).
+`all=false` returns only exported symbols; whereas `all=true` also includes
+non-exported symbols, following the behavior of [`names`](@ref).
+
+See also: [`names`](@ref), [`Docs.hasdoc`](@ref).
+"""
+function undocumented_names(mod::Module; all=false)
+    filter(names(mod; all)) do sym
+        !hasdoc(mod, sym)
+    end
```

Comment:
```suggestion
function undocumented_names(mod::Module; all::Bool=false, identifiers::Bool=true)
    filter!(names(mod; all)) do sym
        !hasdoc(mod, sym) && (!identifiers || Base.isidentifier(sym))
    end
```
Maybe add an option here to filter out symbols that are not valid identifiers by default.